File: generator-common/src/main/java/io/github/potjerodekool/openapi/common/util/StringUtils.java
package io.github.potjerodekool.openapi.common.util;

import static io.github.potjerodekool.codegen.model.util.StringUtils.firstUpper;

public final class StringUtils {

    private StringUtils() {
    }

    public static String toValidClassName(final String name) {
        var className = name;
        className = upperFirst(className.split("-"));
        return upperFirst(className.split("_"));

    }

    private static String upperFirst(final String[] parts) {
        final var sb = new StringBuilder();

        for (final String part : parts) {
            sb.append(firstUpper(part));
        }

        return sb.toString();
    }

    public static String toSnakeCase(final String value) {
        if (value == null || value.isEmpty()) {
            return value;
        } else {
            final var builder = new StringBuilder();

            for (final char c : value.toCharArray()) {
                if (Character.isUpperCase(c)) {
                    builder.append('_');
                    builder.append(Character.toLowerCase(c));
                } else {
                    builder.append(c);
                }
            }
            return builder.toString();
        }
    }

    public static String toUpperCase(final String value) {
        if (value == null || value.isEmpty()) {
            return value;
        } else {
            return value.toUpperCase();
        }
    }
}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/util/StringUtils.kt
package io.github.potjerodekool.openapi.common.util

import io.github.potjerodekool.codegen.model.util.StringUtils
import java.util.*

object StringUtils {
    @JvmStatic
    fun toValidClassName(name: String): String {
        var className = name
        className = upperFirst(className.split("-".toRegex()).dropLastWhile { it.isEmpty() }
            .toTypedArray())
        return upperFirst(className.split("_".toRegex()).dropLastWhile { it.isEmpty() }
            .toTypedArray())
    }

    private fun upperFirst(parts: Array<String>): String {
        val sb = StringBuilder()

        for (part in parts) {
            sb.append(StringUtils.firstUpper(part))
        }

        return sb.toString()
    }

    fun toSnakeCase(value: String?): String? {
        if (value.isNullOrEmpty()) {
            return value
        } else {
            val builder = StringBuilder()

            for (c in value.toCharArray()) {
                if (Character.isUpperCase(c)) {
                    builder.append('_')
                    builder.append(c.lowercaseChar())
                } else {
                    builder.append(c)
                }
            }
            return builder.toString()
        }
    }

    fun toUpperCase(value: String?): String? {
        return if (value.isNullOrEmpty()) {
            value
        } else {
            value.uppercase(Locale.getDefault())
        }
    }
}
